core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -655,6 +655,15 @@ package object matryoshka extends CofreeInstances with FreeInstances {
   def toTree[F[_]: Functor: Foldable]: Algebra[F, Tree[F[Unit]]] =
     x => Tree.Node(x.void, x.toStream)
 
+  /** Replaces all instances of `original` in the structure with `replacement`.
+    *
+    * @group algebras
+    */
+  def substitute[T[_[_]], F[_]](original: T[F], replacement: T[F])(implicit T: Equal[T[F]]):
+      T[F] => T[F] \/ T[F] =
+   tf => if (tf ≟ original) replacement.left else tf.right
```

Comment:
Somehow this doesn't strike me as a primitive. I guess the two halves are something like `T[F] => (T[F] => Option[T[F])` and `toRight` (i.e. `T[F] => (Option[T[F]] => T[F] \/ T[F])`), but I don't think the former is useful on its own so I guess I don't have a suggestion.